platform/security/common/src/main/java/ddf/security/samlp/impl/SamlValidator.java

```diff
@@ -287,6 +288,24 @@ private Request(Builder builder) {
       logoutRequest = (LogoutRequest) builder.xmlObject;
     }
 
+    /**
+     * A SAML LogoutRequest must include at least one SessionIndex, if the logout is initiated by
+     * Session Participant. However, if the Session Authority initiates the logout, a SessionIndex
+     * is optional. In that case, the absence of a SessionIndex means to logout of all sessions
+     * associated with the Principle. Because the SAMLValidator does not have enough information to
+     * determine who initiated the logout, this method does not throw a ValidationException.
+     * Instead, it logs the condition.
+     *
+     * @throws ValidationException
+     */
+    protected void checkSessionIndexes() throws ValidationException {
+      if (logoutRequest.getSessionIndexes().isEmpty()) {
```

Comment:
There isn't really way to test it with the existing unit tests.  I can create a new unit test that to will invoke the method on a mocked LogoutRequest, but I don't feel like it will prove anything.

It would ensure this method does not throw an exception if the collection is empty:

```
    protected void checkSessionIndexes() {
      if (logoutRequest.getSessionIndexes().isEmpty()) {
        LOGGER.info(
            "Logout request does not contain a session index for name-id {}",
            logoutRequest.getNameID());
      }
    }
```
The only different between the two conditions is a logging statement.

Reply:
I agree that both tests would have the same assertion at the end, i.e., no exception is being thrown, but I think that's something that's worth verifying. Up to you.